Specs/Core/CesiumTerrainProviderSpec.js

```diff
@@ -489,16 +494,17 @@ describe("Core/CesiumTerrainProvider", function () {
       url: "made/up/url",
     });
 
-    const deferred = defer();
-
-    provider.errorEvent.addEventListener(function (e) {
-      deferred.resolve(e);
-    });
-
-    return deferred.promise.then(function (error) {
-      expect(error.message).toContain(
+    const errorMatcher = function (event) {
+      expect(event.message).toContain(
         "does not specify any tile URL templates"
       );
+      provider.errorEvent.removeEventListener(errorMatcher);
+    };
+
+    provider.errorEvent.addEventListener(errorMatcher);
+
+    return provider.readyPromise.then(fail).catch((e) => {
+      expect(e.message).toContain("An error occurred while accessing");
```

Comment:
This is starting to get nit-picky, but to cover all our bases in these tests, we should have an expectation to confirm that `errorMatcher` was called after the `readyPromise` is rejected.